### DISCUSSION TITLE
IM getting 58190 illegal hardware instruction when trying to install on  Macbook pro M1  node-llama-cpp

### DISCUSSION THREAD

**Discussion by stevensanborn** [withcatai/node-llama-cpp#401] [General]
pnpm i node-llama-cpp 
Do i not have the proper hardware?
node v20.10.0

Progress: resolved 185, reused 169, downloaded 0, added 0, done
node_modules/.pnpm/node-llama-cpp@3.3.1/node_modules/node-llama-cpp: Running postinstall script, failed in 635ms
.../node_modules/node-llama-cpp postinstall$ node ./dist/cli/cli.js postinstall
└─ Failed in 635ms at /Users/.../Documents/Crypto/eliza/test/node_modules/.pnpm/node-llama-cpp@3.3.1/node_modules/node-llama-cpp
[1]    58190 illegal hardware instruction  pnpm i node-llama-cpp

**Comment by stevensanborn** [comment]
i think m1 is not compatble since m1 has no ability to use AVX2

**Comment by giladgd** [comment]
This error usually happens when running nodejs through Resetta on macOS: https://node-llama-cpp.withcat.ai/guide/troubleshooting#crashes-with-an-illegal-hardware-instruction-error-or-a-sigill-signal

**Comment by stevensanborn** [comment]
nvm install v22.12.0 -arch arm 
that worked thanks
